tests/h/views/api/groups_test.py

```diff
@@ -237,6 +237,18 @@ def test_it_raises_HTTPNotFound_with_non_existent_user(self,
         with pytest.raises(HTTPNotFound):
             views.add_member(group, pyramid_request)
 
+    def test_it_raises_HTTPNotFound_if_userid_malformed(self,
+                                                        group,
+                                                        pyramid_request,
+                                                        user_service,):
+
+        user_service.fetch.side_effect = ValueError('nope')
+
+        pyramid_request.matchdict['userid'] = "invalidformat@wherever"
+
+        with pytest.raises(HTTPNotFound):  # view handles ValueError and raises NotFound
```

Comment:
These kind of tests worry me a little bit because there are multiple reasons why this view might raise HTTPNotFound. It could be that the test set up is wrong and `user_svc.fetch()` is actually returning `None` when the view calls it, which would cause the view to raise HTTPNotFound for a _different_ reason. And then the view code could still be crashing when `fetch()` raises but this test would still be passing.

Having the view raises specific different HTTPNotFound subclasses for the different reasons, and the test catch the correct subclass only, avoids this problem. Or having the view pass an error string to HTTPNotFound and the test match against it.

In this class it seems pretty clear from the test code that the test set up _is_ correct. And it definitely is (I tested). Mistakes are surprisingly each to make, however (e.g. the `user_service` mock itself, whose code is in a different part of the file from the test itself, could be patching the wrong thing and having no effect).

Reply:
This view is up for nearly immediate refactor (to get the other auth'z work I need to do in place), so I will address the Exception-raising stuff at that point (this might not even go in the view at the end of the day).